File: Twopointers/java/ContainerWithMostWater.java
package Twopointers.java;

public class ContainerWithMostWater {
    public int maxArea(int[] height) {
        int left = 0;
        int right = height.length - 1;
        int res = 0;

        while (left < right) {
            int area = (right - left) * Math.min(height[left], height[right]);
            res = Math.max(res, area);

            if (height[left] < height[right]) {
                left++;
            } else {
                right--;
            }
        }

        return res;
    }

}

File: Twopointers/ContainerWithMostWater.scala
object Solution {
  def maxArea(height: Array[Int]): Int = {
    var best = 0
    var i = 0
    var k = height.length - 1

    while (k > i) {
      var area = math.min(height(i), height(k)) * (k - i)
      if (area > best) best = area

      if (height(i) < height(k)) i += 1
      else k -= 1
    }
    best

  }
}
